WordPress/src/main/java/org/wordpress/android/ui/posts/PostsListFragment.java

```diff
@@ -772,7 +772,21 @@ public void onMediaUploaded(OnMediaUploaded event) {
 
         PostModel post = mPostStore.getPostByLocalPostId(event.media.getLocalPostId());
         if (post != null) {
-            mPostsListAdapter.updateProgressForPost(post);
+            if ((event.media.isError() || event.canceled)){
+                // if a media is cancelled or ends in error, and the post is not uploading nor queued,
+                // (meaning there is no other pending media to be uploaded for this post)
+                // then we should refresh it to show its new state
+                if (post != null && !UploadService.isPostUploadingOrQueued(post)) {
+                    // TODO: replace loadPosts for getPostListAdapter().notifyItemChanged(); kind of thing,
```

Comment:
We can clean up this comment a bit - I think even just this TODO is more than enough. #6343 ended up using a different approach, and the link to `PostsListAdapter` is a 404.